src/widgets/canvas/line.rs

```diff
@@ -64,7 +64,7 @@ fn draw_line_low(painter: &mut Painter, x1: usize, y1: usize, x2: usize, y2: usi
     for x in x1..=x2 {
         painter.paint(x, y, color);
         if d > 0 {
-            y = if y1 > y2 { y - 1 } else { y + 1 };
+            y = if y1 > y2 && y1 > 0 { y - 1 } else { y + 1 };
```

Comment:
what about the *theoretical* overflow case of `y+1` ? would it be clearer communication-wise to use `saturated_add`/`saturated_sub` here?

Reply:
I personally also think just using `saturated_sub/saturated_add` will do the trick.

Reply:
TIL these exist. I'll update after I finish up after work. Thanks for the suggestions!

Reply:
Saturated!